### PR TITLE
Dismiss vulnerabilities

### PR DESCRIPTION
**Pull request by jclark-vmware**
## Pull request checklist

Please check if your PR fulfills the following requirements:

- [x] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [ ] Tests have been added that prove the fix is effective or that the feature works.
- [ ] New and existing tests pass locally with the changes.
- [ ] Code follows the style guidelines of this project (PEP8, clean code, linter).
- [ ] A self-review of the code has been done.

## What is the ticket or issue number?

CBAPI-4238

## Pull Request Description

Partial work for dismiss, undismiss, and dismiss edit of vulnerabilities. Not complete or tested; checking in due to illness so it can be picked up

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: src/cbc_sdk/platform/vulnerability_assessment.py [hunk 1]
```diff
@@ -104,6 +104,50 @@ def _query_implementation(cls, cb, **kwargs):
         """
         return VulnerabilityQuery(cls, cb)
 
+    def perform_action(self, type, reason=None, notes=None):
+        """
+        Take an action on to manage the Vulnerability.
```
Comment (kebringer-cb) [inline src/cbc_sdk/platform/vulnerability_assessment.py:109]: Think this should be one of:
Take an action on the Vulnerability.
Take an action to manage the Vulnerability.

File: src/cbc_sdk/platform/vulnerability_assessment.py [hunk 2]
```diff
@@ -815,6 +938,28 @@ def _run_async_query(self, context):
             if current >= self._total_results:
                 return self._doc_class(self._cb, initial_data=results)
 
+    def export(self):
```
Comment (kebringer-cb) [inline src/cbc_sdk/platform/vulnerability_assessment.py:941]: Would be good to have an example snippet of what to do with the returned object.  Either in line here, or added to the vuln guide.

File: src/cbc_sdk/platform/vulnerability_assessment.py [hunk 3]
```diff
@@ -211,6 +257,21 @@ def set_vcenter(self, vcenter_uuid):
...
 
+    def set_visibility(self, visibility):
+        """
+        Restricts the vulnerabilities that this query is performed on to the specified visibility
+
+        Args:
+            visibility (str): The visibility state of the vulnerabilty. (supports ACTIVE, ALL, DISMISSED)
+
+        Returns:
+            VulnerabilityQuery: This instance.
```
Comment (emitreva-cb) [inline src/cbc_sdk/platform/vulnerability_assessment.py:268]: Maybe this should be returning VulnerabilityOrgSummaryQuery.

File: src/cbc_sdk/platform/vulnerability_assessment.py [hunk 4]
```diff
@@ -320,6 +401,22 @@ def add_criteria(self, key, value, operator='EQUALS'):
...
+        Args:
+            deployment_type (str): deployment type ("ENDPOINT", "AWS")
+            operator (str):  logic operator to apply to property value.
+
+        Returns:
+            VulnerabilityQuery: This instance.
+        """
+        if not deployment_type:
+            raise ApiError("Invalid deployment type")
+        self._update_criteria("deployment_type", deployment_type, operator)
```
Comment (emitreva-cb) [inline src/cbc_sdk/platform/vulnerability_assessment.py:417]: Should we have some validation on deployment_types or is not needed?

File: src/cbc_sdk/platform/vulnerability_assessment.py [hunk 5]
```diff
@@ -832,6 +977,22 @@ def __init__(self, vulnerability, cb):
...
+    def set_os_product_id(self, os_product_id, operator):
+        """
+        Restricts the vulnerabilities that this query is performed on to the specified os_product_id.
+
+        Args:
+            os_product_id (str): os_product_id.
+            operator (str):  logic operator to apply to property value.
+
+        Returns:
+            VulnerabilityQuery: This instance.
```
Comment (emitreva-cb) [inline src/cbc_sdk/platform/vulnerability_assessment.py:989]: Wrong return type.

File: src/cbc_sdk/platform/vulnerability_assessment.py (hunk 3 above)
Comment (abowersox-cb) [reply src/cbc_sdk/platform/vulnerability_assessment.py:268]: Yes, it should.

File: src/cbc_sdk/platform/vulnerability_assessment.py (hunk 5 above)
Comment (abowersox-cb) [reply src/cbc_sdk/platform/vulnerability_assessment.py:989]: Yes, should be `AffectedAssetQuery`.

File: src/cbc_sdk/platform/vulnerability_assessment.py (hunk 4 above)
Comment (avanbrunt-cb) [reply src/cbc_sdk/platform/vulnerability_assessment.py:417]: I'm hesitant to add this with all the public cloud items coming out. The API will respond with an error anyways so it only saves us an API call

**Review by emitreva-cb — APPROVED** [review]

**Review by abowersox-cb — APPROVED** [review]

**Review by kebringer-cb — APPROVED** [review]

**Pull request merged**